File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        left = maxLen = maxCount = 0
        freq = [0] * 26
        for right in range(len(s)):
            freq[ord(s[right]) - ord('A')] += 1
            maxCount = max(maxCount, freq[ord(s[right]) - ord('A')])
            if (right - left + 1) - maxCount > k:
                freq[ord(s[left]) - ord('A')] -= 1
                left += 1
            maxLen = max(maxLen, right - left + 1)
        return maxLen

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int left = 0, maxLen = 0, n = s.size(), maxCount = 0;
        int freq[26] = {0};
        for(int right = 0; right < n; right++){
            freq[s[right] - 'A']++;
            maxCount = max(maxCount, freq[s[right] - 'A']);
            if((right - left + 1) - maxCount> k){
                freq[s[left] - 'A']--;
                left++;
            }
            maxLen = max(maxLen, right - left + 1);
        }
        return maxLen;
    }
};